bbc_microbit_v2 build error for lis2dh

[lis2dh_output.txt](https://github.com/zephyrproject-rtos/zephyr/files/5996823/lis2dh_output.txt)

**Describe the bug**
Error while building the example lis2dh, it works for bbc_microbit but not for bbc_microbit_v2.
I have tried to find the issue, but I have no luck, one thing which might cause some problems is that the `sm303agr-magn@1e` and `lsm303agr-accel@19` from the `bbc_microbit_v2.dts` file are on the same GPIO pin.

![image](https://user-images.githubusercontent.com/38521446/108223799-16d08580-7132-11eb-9b55-94becea58dd8.png)


**To Reproduce**
Steps to reproduce the behavior:
1. zephyr 2.4
2. cd zephyr/samples
3. west build -b bbc_microbit_v2 samples/sensor/lis2dh
4. See the error

**Expected behavior**
Bbc_microbit (v1.5 and v2) share the same accelerometer **lsm303agr** it is expected for the same code to work, or a solution to access the accelerometer for v2. 

**Impact**
Not able to progress on my FYP (Final year project) because I can not access the accelerometer information.

**Logs and console output**
 ```
In file included from ../include/sys/util.h:27,
                 from ../include/sched_priq.h:9,
                 from ../include/kernel_includes.h:23,
                 from ../include/kernel.h:17,
                 from ../include/init.h:11,
                 from /home/liviu/Part2/zephyrproject/zephyr/drivers/sensor/lis2dh/lis2dh.c:10:

zephyr/include/generated/devicetree_unfixed.h:2926:40: error: 'DT_N_S_soc_S_i2c_40003000_S_lsm303agr_accel_19_BUS_P_label' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_i2c_40003000_S_lsm303agr_accel_19_P_label'?
 2926 | #define DT_N_INST_0_st_lis2dh          DT_N_S_soc_S_i2c_40003000_S_lsm303agr_accel_19
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../include/sys/util_internal.h:65:26: note: in definition of macro '__DEBRACKET'
   65 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
../include/sys/util_internal.h:57:2: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   57 |  __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |  ^~~~~~~~~~~~~~~~~~~~
../include/sys/util_internal.h:52:2: note: in expansion of macro '__COND_CODE'
   52 |  __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |  ^~~~~~~~~~~
../include/sys/util.h:414:2: note: in expansion of macro 'Z_COND_CODE_1'
  414 |  Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |  ^~~~~~~~~~~~~
```
There are 300 lines, I have attached the file lis2dh_output.txt for the full output.

**Environment (please complete the following information):**
 - OS: Linux 20.04
 - Toolchain: Zephyr version: 2.4.0, west 0.9


**Additional context**
The bbc_microbit (v1.5) has the Cortex-M0 processor while v2 has the Cortex-M4 processor, I am not sure if that should make any difference in the ability to use the example lis2dh.

--- a/boards/arm/bbc_microbit_v2/bbc_microbit_v2.dts
+++ b/boards/arm/bbc_microbit_v2/bbc_microbit_v2.dts
@@ -88,6 +88,7 @@
 };
 
 &i2c0 {
+	compatible = "nordic,nrf-twim";
 	status = "okay";
 	clock-frequency = <I2C_BITRATE_FAST>;
 	sda-pin = <16>;

